### ISSUE TITLE
buildValidation error in PHP 5.3

### ISSUE THREAD

**Issue by markitosgv** [symfony/symfony#14559]
Im using Symfony 2.6 and when I try to execute custom validation throw this error:
```
request.CRITICAL: Uncaught PHP Exception Symfony\Component\Debug\Exception\UndefinedMethodException: "Attempted to call method "buildViolation" on class "Symfony\Component\Validator\ExecutionContext". Did you mean to call "addViolation"?"
```

My code is:

```php
$this->context->buildViolation($constraint->message)
                    ->setParameter('%text%', $text)
                    ->addViolation();
```

Only occurs in production server with PHP 5.3.3 (CentOS), on local machine with Ubuntu and PHP 5.5 works great...

**Comment by dosten** [comment]
What is your exact version? v2.6.x

**Comment by dosten** [comment]
You are using the old API your code must be:
```
$this->buildViolation($constraint->message)
                    ->setParameter('%text%', $text)
                    ->addViolation();
```

**Comment by stof** [comment]
@markitosgv on PHP 5.3.3, the 2.5-BC validation API is not available (it is not compatible with PHP 5.3.8 and below). For such versions, the default configuration is to enable the 2.4 API rather than the 2.5 API (enabling only the 2.5 API is a BC break). And there is no ``buildViolation`` method in the old API.

If your code is deployed to PHP 5.3.3, you should force the validation layer to use the 2.4 API version only even locally (and your CI server should run 5.3.3 too, to catch it earlier than the production). However, be careful that Symfony 2.7 does not support PHP 5.3.3 anymore. the min requirement has been bumped to 5.3.9 (this issue is one of the reasons). So you should consider upgrading to a newer PHP version

**Comment by markitosgv** [comment]
Ok, then i think the best option is upgrade PHP version in prod Centos. Thanks a lot for explanation @stof
